src/Handler/DoctrineNullablePropertyHandler.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Psalm\SymfonyPsalmPlugin\Handler;
+
+use Doctrine\Common\Annotations\AnnotationException;
+use Doctrine\Common\Annotations\AnnotationReader;
+use Doctrine\ORM\Mapping\Column;
+use PhpParser\Node\Stmt\ClassLike;
+use Psalm\Codebase;
+use Psalm\CodeLocation;
+use Psalm\FileSource;
+use Psalm\IssueBuffer;
+use Psalm\Plugin\Hook\AfterClassLikeVisitInterface;
+use Psalm\Storage\ClassLikeStorage;
+use Psalm\SymfonyPsalmPlugin\Issue\DoctrinePropertyInvalidType;
+
+class DoctrineNullablePropertyHandler implements AfterClassLikeVisitInterface
+{
+    public static function afterClassLikeVisit(
+        ClassLike $stmt,
+        ClassLikeStorage $storage,
+        FileSource $statements_source,
+        Codebase $codebase,
+        array &$file_replacements = []
+    ): void {
+        $className = $storage->name;
+        try {
+            $reflectionClass = new \ReflectionClass($className);
+        } catch (\ReflectionException $e) {
+            return;
+        }
+        $annotationReader = new AnnotationReader();
+
+        foreach ($stmt->getProperties() as $property) {
+            try {
+                $propertyReflection = $reflectionClass->getProperty($property->props[0]->name->name);
+                $columnAnnotation = $annotationReader->getPropertyAnnotation($propertyReflection, Column::class);
+            } catch (\ReflectionException $e) {
+                continue;
+            } catch (AnnotationException $e) {
+                continue;
+            }
+
+            if (!$columnAnnotation) {
+                continue;
+            }
+
+            if (!isset($storage->properties[$propertyReflection->name])) {
+                continue;
+            }
+
+            $propertyInfo = $storage->properties[$propertyReflection->name];
+            if (!$propertyInfo->type) {
+                continue;
+            }
+
+            $nullableByDoctrine = $columnAnnotation->nullable;
+
+            if ($propertyInfo->type->isNullable() && !$nullableByDoctrine) {
```

Comment:
I agree with the issue for
```
/**
 * @var string
 * @ORM(type=string, nullable=true)
 */
```

But I don't think you can have an issue for
```
/**
 * @var string|null
 * @ORM(type=string, nullable=false)
 */
```

Because when creating a new object `new Entity()`, the property is null at first, so it should be an authorized type.

Reply:
```php
/**
 * @var string|null
 * @ORM(type=string, nullable=false)
 */
```
in this case, objects can be created with required values in construct `new Entity('string', ...)`. Nevertheless this approach may not fit for every project, so maybe enabling/disabling this feature should be configurable.

Reply:
> Nevertheless this approach may not fit for every project, so maybe enabling/disabling this feature should be configurable.

I can confirm it won't fit with my project indeed. A configurable feature seems perfect then :)

Reply:
The same check exists for phpstan, and an option `allowNullablePropertyForRequiredField` was provided 
https://github.com/phpstan/phpstan-doctrine/pull/208

Also, shouldn't this be in the Doctrine plugin rather than the symfony one ?